src/core/src/op/convert_align_types.cpp

```diff
@@ -0,0 +1,171 @@
+    const auto& rhs_type = rhs.get_element_type();
+    NODE_VALIDATION_CHECK(op,
+                          std::find(supported_types.begin(), supported_types.end(), lhs_type) != supported_types.end());
+    NODE_VALIDATION_CHECK(op,
+                          std::find(supported_types.begin(), supported_types.end(), rhs_type) != supported_types.end());
+    if (lhs_type.is_dynamic() || rhs_type.is_dynamic()) {
+        return element::dynamic;
+    }
+    if (lhs_type == rhs_type)
+        return lhs_type;
+    if (lhs_type == element::boolean)
+        return rhs_type;
+    if (rhs_type == element::boolean)
+        return lhs_type;
+    const auto& lhs_rank = lhs.get_partial_shape().rank();
+    const auto& rhs_rank = rhs.get_partial_shape().rank();
+    const bool is_lhs_scalar = lhs_rank.is_static() ? lhs_rank.get_length() == 0 : false;
+    const bool is_rhs_scalar = rhs_rank.is_static() ? rhs_rank.get_length() == 0 : false;
+    const bool is_lhs_signed = lhs_type.is_signed();
+    const bool is_rhs_signed = rhs_type.is_signed();
+    const bool is_lhs_real = lhs_type.is_real();
+    const bool is_rhs_real = rhs_type.is_real();
+    const size_t lhs_bitwidth = lhs_type.bitwidth();
+    const size_t rhs_bitwidth = rhs_type.bitwidth();
+
+    if (pytorch_scalar_align && (is_lhs_scalar ^ is_rhs_scalar) && (is_lhs_real == is_rhs_real)) {
+        if (promote_unsafe) {
+            return is_lhs_scalar ? rhs_type : lhs_type;
+        }
+        const auto target = is_lhs_scalar ? rhs_type : lhs_type;
+        const auto scalar = is_lhs_scalar ? lhs_type : rhs_type;
+        if ((target.is_signed() == scalar.is_signed() && target.bitwidth() >= scalar.bitwidth()) ||
+            (target.is_signed() && !scalar.is_signed() && target.bitwidth() * 2 >= scalar.bitwidth())) {
+            return target;
+        }
+        NODE_VALIDATION_CHECK(op, false, " Scalar input cannot be PyTorch-like aligned using safe promotion rules.");
+    }
+    if (is_lhs_real ^ is_rhs_real) {
+        if (promote_unsafe) {
+            return is_lhs_real ? lhs_type : rhs_type;
+        }
+        const auto real = is_lhs_real ? lhs_type : rhs_type;
+        const auto integer = is_lhs_real ? rhs_type : lhs_type;
+        if (real.bitwidth() >= integer.bitwidth() * 2) {
+            return real;
+        }
+        NODE_VALIDATION_CHECK(op, false, "Integer input cannot be safely promoted to floating-point.");
+    }
+    if ((is_lhs_real == is_rhs_real) && (is_lhs_signed ^ is_rhs_signed)) {
+        const auto uint_bitwidth = is_lhs_signed ? rhs_bitwidth : lhs_bitwidth;
+        const auto int_bitwidth = is_lhs_signed ? lhs_bitwidth : rhs_bitwidth;
+        if (uint_bitwidth <= 32 && (promote_unsafe || uint_bitwidth * 2 <= int_bitwidth)) {
+            return bit_to_int.at(std::max({uint_bitwidth * 2, int_bitwidth}));
+        } else if (promote_unsafe) {
+            return u64_promotion_target;
+        } else {
+            NODE_VALIDATION_CHECK(
```

Comment:
In this kind of errors would be nice to know which input has wrong type.

Reply:
If something fails, technically you can say that both inputs have wrong datatypes if alignment to either one of them is not possible :)
In this case either u64 would need to be u64 or second input would need to be unsigned.
Or did you mean to introduce some logic to give some suggestions what changes to inputs are needed (like increase bitwidth, change number type) to allow it to work un safe mode?